.github/workflows/platform-autogpt-deploy.yaml

```diff
@@ -19,8 +19,53 @@ env:
   NAMESPACE: dev-agpt
 
 jobs:
+  migrate:
+    environment: develop
+    name: Run migrations for AutoGPT Platform
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install prisma
+
+      - name: Check for changes
+        id: check_changes
+        run: |
+          git fetch origin dev
+          BACKEND_CHANGED=$(git diff --name-only origin/dev HEAD | grep "^autogpt_platform/backend/migrations" && echo "true" || echo "false")
+          MARKET_CHANGED=$(git diff --name-only origin/dev HEAD | grep "^autogpt_platform/market/migrations" && echo "true" || echo "false")
+          echo "backend_changed=$BACKEND_CHANGED" >> $GITHUB_OUTPUT
+          echo "market_changed=$MARKET_CHANGED" >> $GITHUB_OUTPUT
+
+      - name: Run Backend Migrations
+        if: steps.check_changes.outputs.backend_changed == 'true'
+        working-directory: ./autogpt_platform/backend
+        run: |
+          python -m prisma migrate deploy
+        env:
+          DATABASE_URL: ${{ secrets.BACKEND_DATABASE_URL }}
```

Comment:
I'm not the smartest bird in the bush so forgive me if this is wild but wont that be the same as the prod one?

Reply:
@ntindle https://github.com/Significant-Gravitas/AutoGPT/pull/8395/files/dc5a46b8a37656a23ea0accde00d5857c301272b#diff-cdb7aebf544b4e121ce5e18aaaf57650cdc6f9a3ce47f54681fcf695aba66a03

I'm using gh environments, both envvars are different